app/code/Magento/InventoryIndexer/Indexer/Stock/PrepareIndexDataForClearingIndex.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryIndexer\Indexer\Stock;
+
+use ArrayIterator;
+
+/**
+ * Class PrepareIndexDataForClearingIndex
+ */
+class PrepareIndexDataForClearingIndex
+{
+    /**
+     * Deletes all unnecessary data from indexData for clearing
+     *
+     * @param ArrayIterator $indexData
+     *
+     * @return ArrayIterator
+     */
+    public function execute(ArrayIterator $indexData): ArrayIterator
+    {
+        $clearIndex = [];
+        foreach ($indexData as $indexItem) {
+            $clearIndex[]['sku'] = $indexItem['sku'];
```

Comment:
Can we use `[array_column](https://www.php.net/manual/en/function.array-column.php)`?

Reply:
No. Returning structure will be different.